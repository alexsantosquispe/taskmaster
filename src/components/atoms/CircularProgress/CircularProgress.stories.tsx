import { Wrapper } from '@/utils/wrappers.utils';
import type { Meta, StoryObj } from '@storybook/react';
import { CircularProgress } from './CircularProgress';

const meta = {
  title: 'Atoms/CircularProgress',
  component: CircularProgress,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof CircularProgress>;

export default meta;

type Story = StoryObj<typeof CircularProgress>;

export const Default: Story = () => {
  return (
    <div className="flex gap-8">
      <Wrapper className="min-w-24">
        <CircularProgress percentage={50} />
      </Wrapper>
      <Wrapper className="min-w-24">
        <CircularProgress percentage={80} size={30} strokeWidth={3} />
      </Wrapper>
    </div>
  );
};

Default.args = {};
