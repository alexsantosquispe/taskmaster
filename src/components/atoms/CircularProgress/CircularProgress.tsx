import { twMerge } from 'tailwind-merge';

interface CircularProgressProps {
  percentage: number;
  size?: number;
  strokeWidth?: number;
  className?: string;
}

const INITIAL_PERCENTAGE = 1;
const DEFAULT_SIZE = 16;
const STROKE_WIDTH = 2;

export const CircularProgress = ({
  percentage,
  size = DEFAULT_SIZE,
  strokeWidth = STROKE_WIDTH,
  className
}: CircularProgressProps) => {
  const defaultPercentage = percentage || INITIAL_PERCENTAGE;
  const radius = (size - strokeWidth) / 2;
  const circumference = 2 * Math.PI * radius;
  const offset = circumference * (1 - defaultPercentage / 100);

  return (
    <svg width={size} height={size}>
      <circle
        cx={size / 2}
        cy={size / 2}
        r={radius}
        className="stroke-gray-200 dark:stroke-neutral-600"
        strokeWidth={strokeWidth}
        fill="none"
      />
      <circle
        cx={size / 2}
        cy={size / 2}
        r={radius}
        strokeWidth={strokeWidth}
        fill="none"
        strokeDasharray={circumference}
        strokeDashoffset={offset}
        strokeLinecap="round"
        transform={`rotate(-90 ${size / 2} ${size / 2})`}
        className={twMerge(className)}
      />
    </svg>
  );
};
