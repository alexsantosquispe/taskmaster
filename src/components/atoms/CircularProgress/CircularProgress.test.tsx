import { render } from '@testing-library/react';
import { CircularProgress } from './CircularProgress';

describe('CircularProgress', () => {
  it('should render the component correctly', () => {
    const component = render(<CircularProgress percentage={0} />);

    expect(component).toMatchSnapshot();
  });
});
