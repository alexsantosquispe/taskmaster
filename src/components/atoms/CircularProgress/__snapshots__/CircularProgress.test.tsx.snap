// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CircularProgress should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        height="16"
        width="16"
      >
        <circle
          class="stroke-gray-200 dark:stroke-neutral-600"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-width="2"
        />
        <circle
          class="stroke-primary dark:stroke-white"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-dasharray="43.982297150257104"
          stroke-dashoffset="43.54247417875453"
          stroke-linecap="round"
          stroke-width="2"
          transform="rotate(-90 8 8)"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      height="16"
      width="16"
    >
      <circle
        class="stroke-gray-200 dark:stroke-neutral-600"
        cx="8"
        cy="8"
        fill="none"
        r="7"
        stroke-width="2"
      />
      <circle
        class="stroke-primary dark:stroke-white"
        cx="8"
        cy="8"
        fill="none"
        r="7"
        stroke-dasharray="43.982297150257104"
        stroke-dashoffset="43.54247417875453"
        stroke-linecap="round"
        stroke-width="2"
        transform="rotate(-90 8 8)"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
